jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -670,14 +667,33 @@ YUI.add('user-profile', function() {
     },
 
     /**
+      Handle the destroy model button being clicked.
+
+      @method _handleDestroyModel
+    */
+    _handleDestroyModel: function() {
+      var model = this.state.destroyModel;
+      var uuid = model.uuid;
+      // Hide the confirmation popup.
+      this._displayConfirmation(null);
+      // If the model to be distroyed is active then we can destroy it
+      // immediately.
+      if (uuid === this.props.currentModel) {
```

Comment:
This is very nice!
The only thing I am not sure about is error handling here. Is it possible that switchModel returns an error before calling the callback? What happens if switchEnv fails (for instance because the corresponding model is being removed in the meanwhile) and "login" is never fired? I am not sure it's easy to handle these cases, and surely you don't have to do that in this branch. But maybe add a TODO here to investigate possible solutions later?